lib/pages/team_info_form.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:dropdown_search/dropdown_search.dart';
+import 'package:flutter/material.dart';
+import 'package:neatteam_scouting_2023/models/game.dart';
+import 'package:neatteam_scouting_2023/models/team.dart';
+import 'package:neatteam_scouting_2023/styles/style_form_field.dart';
+import 'package:neatteam_scouting_2023/utils/frc_teams.dart';
+
+class TeamInfoFormPage extends StatefulWidget {
+  const TeamInfoFormPage({super.key});
+
+  @override
+  State<StatefulWidget> createState() => _TeamInfoForm();
+}
+
+class _TeamInfoForm extends State<TeamInfoFormPage> {
+  final _formKey = GlobalKey<FormState>();
+
+  final _matchController = TextEditingController();
+  Team? _selectedTeam;
+  String? _selectedAlliance;
+
+  List<Team> _teams = [];
+
+  @override
+  void initState() {
+    super.initState();
+    FrcTeams.getAllTeams().then((teams) => setState(() => _teams = teams));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Team info')),
+      body: Form(
+          key: _formKey,
+          child: Column(
+            children: [
+              StyleFormField(
+                  field: TextFormField(
+                controller: _matchController,
+                validator: (value) {
+                  if (value == null || value.isEmpty) {
+                    return 'Please enter match number!';
+                  } else if (int.tryParse(value) == null) {
+                    return 'Please enter a numeric value!';
+                  }
+                  return null;
+                },
+                keyboardType: TextInputType.number,
+                decoration: const InputDecoration(labelText: "Match"),
+              )),
+              StyleFormField(
+                  field: SingleChildScrollView(
+                      child: DropdownSearch<Team>(
+                          items: _teams,
+                          filterFn: (Team team, String filter) {
+                            return team.name.toLowerCase().contains(filter.toLowerCase()) ||
+                                team.number.toString().contains(filter);
```

Comment:
Redundant check, transforming `AeTa` to lower results in `aeta`, doing it for both sides of the comparison eliminates the need in further checks since letter case won't change your equality anymore.

Reply:
replaced with a regex search